types/signature.go

```diff
@@ -0,0 +1,72 @@
+package types
+
+import (
+	"crypto/ed25519"
+	"encoding/base64"
+	"strings"
+	"time"
+)
+
+// Cleartext format:
+// Matt Butcher <matt.butcher@example.com>
+// mybindle
+// 0.1.0
+// creator
+// ~
+// e1706ab0a39ac88094b6d54a3f5cdba41fe5a901
+// 098fa798779ac88094b6d54a3f5cdba41fe5a901
+// 5b992e90b71d5fadab3cd3777230ef370df75f5b
+
+// NOTE: the spec (https://github.com/deislabs/bindle/blob/main/docs/signing-spec.md#signing-on-the-invoice)
+// includes the `at` value in the cleartext, but the server does not, so this client does not either.
+// Issue: https://github.com/deislabs/bindle/issues/284
+
+// GenerateCreatorSignature generates a signature for the creator using the first 'author'
+// in the Invoice, and then appends the new signature to the Invoice's 'Signature' list.
+// Use keyring.GenerateSignatureKey to create a keypair
+func (i *Invoice) GenerateCreatorSignature(sigKey *SignatureKey, privKey []byte) error {
+	timestamp := time.Now()
+
+	cleartext := i.generateCleartext("creator", timestamp)
+
+	sig := ed25519.Sign(privKey, []byte(cleartext))
```

Comment:
This should have a check to make sure the incoming signatureKey has the `creator` role in its list

Reply:
Is this check still required if I change the method to accept any author/role combo?

Reply:
Answered my own question, will just use the `role` on the SignatureKey, duh :)

Reply:
Or at least check to see that the given role is included in the key's list of roles :P